borg/hashindex.pyx

```diff
@@ -22,13 +22,32 @@ cdef extern from "_hashindex.c":
     void *hashindex_next_key(HashIndex *index, void *key)
     int hashindex_delete(HashIndex *index, void *key)
     int hashindex_set(HashIndex *index, void *key, void *value)
-    int _htole32(int v)
-    int _le32toh(int v)
+    uint32_t _htole32(uint32_t v)
+    uint32_t _le32toh(uint32_t v)
 
 
 cdef _NoDefault = object()
 
-cimport cython
+"""
+The HashIndex is *not* a general purpose data structure. The value size must be at least 4 bytes, and these
+first bytes are used for in-band signalling in the data structure itself (byte ordering is LE)::
+
+    0xffffffff marks empty entries.
+    0xfffffffe marks deleted entries.
+
+This, among other things, limits the number of segments in a repository to 2**32-2. This was a hard limit above
+which data corruption will always occur.
+
+The constant MAX_VALUE is defined as 2**32-1023. As noted above 2**32-1 and -2 are reserved for internal
+use by the HashIndex itself. The values from MAX_VALUE (exclusive) to 2**32-2 (exclusive) are reserved
+as well for future use, should the need for special in-band signalling arise.
+"""
+
+assert UINT32_MAX == 2**32-1
+
+# module-level constant because cdef's in classes can't have default values
+cdef uint32_t _MAX_VALUE = 2**32-1023
```

Comment:
also, if the unit tests rely on it being a odd number, add a comment and a assert here?